### ISSUE TITLE
Add support for refactoring the entity IRI

### ISSUE THREAD

**Issue by tudorache** [protegeproject/webprotege#399]
Currently, the "Show IRI" context-menu item shows the entity IRI in a text field, and it allows to edit the value, but the entity IRI is not actually changed after clicking on OK.

WebProtege should support the refactoring of IRIs, similar to Protege desktop.

**Comment by matthewhorridge** [comment]
This isn't the intended functionality of the "Show IRI" dialog.  It could be....

I need to think about entity IRI changes because we would need to track IRI changes so that comments and history aren't lost (we could update IRIs for comments, but with history it's not so easy)

**Comment by csnyulas** [comment]
This could be especially useful if the IRI are meaningful. At the moment if someone makes a mistake, they need to manually refactor the ontology (create a new entity, move all description from the old to the new, and delete the old entity) which can be really painful.

Actually I was looking for a "Rename class" or similar menu item in the Class Hierarchy portlet, when I realized that the only way to rename classes is to change their `rdfs:label` and there is no way to change their IRI. I guess the "Show IRI" could be used also for changing the IRI, but until this is not the case the text field displaying the IRI should not be editable (see #434).

**Comment by aldogangemi** [comment]
> This could be especially useful if the IRI are meaningful. At the moment if someone makes a mistake, they need to manually refactor the ontology (create a new entity, move all description from the old to the new, and delete the old entity) which can be really painful.
> 
> Actually I was looking for a "Rename class" or similar menu item in the Class Hierarchy portlet, when I realized that the only way to rename classes is to change their `rdfs:label` and there is no way to change their IRI. I guess the "Show IRI" could be used also for changing the IRI, but until this is not the case the text field displaying the IRI should not be editable (see #434).

Hi all, any news on this? The lack of a renaming functionality is utterly annoying, especially in collaborative projects :)
Thanks for your great work
Aldo

**Comment by damienriehlfc** [comment]
I agree with the original poster — and for @aldogangemi 's comment that WebProtege's inability to rename is currently precluding needed functionality. Some examples: 
1. User makes a mistake, then needs to remedy. (As discussed above).
2. User wants to incorporate IRIs from other systems (e.g., desktop-based Protege, other ontologies)

When will WebProtege enable users edit the IRI?  

And I agree: thank you for all of your great work on this amazing software. It's really beautiful.

**Comment by matuskalas** [comment]
I do very much agree with all the posts. 👍🏽

And I understand that the tracking of history is what makes it non-trivial.
(How does the offline Protégé do it?)

However, the lack of the IRI editing option makes WebProtégé severely limited.

So, could we please please 🙏🏽 get the IRI editting in place?

**Comment by abayomiAkanji** [comment]
Hi all, a way to simply solve this issue is to open the ontology file from a text editor and _replace_ the **IRI** with the new one. So, just find text (old name) and replace - using your editor's search feature (e.g. CTR F + CTR R).
_N.B:_ Don't forget to keep a copy of your file (in case you mess up the file and need to revert).

I hope that helps.

**Comment by matuskalas** [comment]
Hi @abayomiAkanji, this is indeed what we're doing.
Unfortunately it isn't satisfactory, as we then loose all history and all comments for that Class/_etc._ in WebProtégé 😢

**Comment by rodrigo-guinea** [comment]
I am looking into applications to design an manage the ontologies of the company I work for. It is a shame, since it is kind of Blocking.

**Comment by mireiarmirapeix** [comment]
Any update on this? It's still not possible to edit IRIs using the web editor.
